MetaMorpheus/GuiFunctions/MetaDraw/SpectrumMatch/SpectrumMatchPlot.cs

```diff
@@ -461,53 +488,67 @@ protected void AnnotateProperties()
                 }
                 else
                     text.Append(SpectrumMatch.ProteinName);
+
                 text.Append("\r\n");
             }
+
             if (MetaDrawSettings.SpectrumDescription["Decoy/Contaminant/Target: "])
             {
                 text.Append("Decoy/Contaminant/Target: ");
                 text.Append(SpectrumMatch.DecoyContamTarget);
                 text.Append("\r\n");
             }
+
             if (MetaDrawSettings.SpectrumDescription["Sequence Length: "])
             {
                 text.Append("Sequence Length: ");
                 text.Append(SpectrumMatch.BaseSeq.Length.ToString("F3").Split('.')[0]);
                 text.Append("\r\n");
             }
+
             if (MetaDrawSettings.SpectrumDescription["Ambiguity Level: "])
             {
                 text.Append("Ambiguity Level: ");
                 text.Append(SpectrumMatch.AmbiguityLevel);
                 text.Append("\r\n");
             }
+
             if (MetaDrawSettings.SpectrumDescription["Spectral Angle: "])
             {
-                text.Append("Spectral Angle: ");
                 if (SpectrumMatch.SpectralAngle != null)
-                    text.Append(SpectrumMatch.SpectralAngle.ToString());
-                else
-                    text.Append("N/A");
-                text.Append("\r\n");
+                {
+                    text.Append("Original Spectral Angle: ");
+                    text.Append(SpectrumMatch.SpectralAngle.ToString() + "\r\n");
+                }
+
+                if (librarySpectrum != null)
```

Comment:
This will calculate the angle on the fly every time a library spectrum is present. Is this the desired behavior or should it be an else if so that it is only calculated when the spectral angle is not present in the psm?

Reply:
it should alwasy happpen. because it is against the loaded library not necessarily with the original library used in the search